src/ParamPassing.hh

```diff
@@ -81,6 +81,13 @@ namespace sdf
     ElementPtr getElementByName(const ElementPtr _elem,
                                 const tinyxml2::XMLElement *_xml);
 
+    /// \brief Initialize an sdf element from xml
+    /// \param[in] _xml Pointer to xml element
+    /// \param[out] _errors Captures errors found during parsing
+    /// \return ElementPtr to the initialized element pointer, nullptr if
+    /// undefined/unknown sdf element
+    ElementPtr initElement(const tinyxml2::XMLElement *_xml, Errors &_errors);
```

Comment:
The function actually initializes the element description which is needed for `readXml`. I've renamed `initElement` to be `initElementDescription` to avoid confusion: e86074c196e3569c6e027e5e7d7477d97dd9ba18